src/planner/expand_hypertable.c

```diff
@@ -660,31 +693,34 @@ process_quals(Node *quals, CollectQualCtx *ctx, bool is_outer_join)
 			Expr *left = linitial(op->args);
 			Expr *right = lsecond(op->args);
 
-			/*
-			 * check for constraints with TIMESTAMPTZ OP INTERVAL calculations
-			 */
 			if ((IsA(left, Var) && is_timestamptz_op_interval(right)) ||
 				(IsA(right, Var) && is_timestamptz_op_interval(left)))
-				qual = (Expr *) constify_timestamptz_op_interval(ctx->root, op);
-
-			/*
-			 * check for time_bucket comparisons
-			 * time_bucket(Const, time_colum) > Const
-			 */
-			if ((IsA(left, FuncExpr) && IsA(right, Const) &&
-				 list_length(castNode(FuncExpr, left)->args) == 2 &&
-				 is_time_bucket_function(left)) ||
-				(IsA(left, Const) && IsA(right, FuncExpr) &&
-				 list_length(castNode(FuncExpr, right)->args) == 2 &&
-				 is_time_bucket_function(right)))
 			{
-				qual = (Expr *) transform_time_bucket_comparison(ctx->root, op);
 				/*
-				 * if we could transform the expression we add it to the list of
-				 * quals so it can be used as an index condition
+				 * check for constraints with TIMESTAMPTZ OP INTERVAL calculations
+				 */
+				qual = &constify_timestamptz_op_interval(ctx->root, op)->xpr;
```

Comment:
why is this changed from `qual = (Expr *) constify_timestamptz_op_interval(ctx->root, op);`

Reply:
I think it's a leftover from back and forth changes, I'll remove it (it's the same cast just written differently).